### PR TITLE
 Epinio: Application UI Improvements

### PR DESCRIPTION
**Pull request by scures**
Recreation of https://github.com/rancher/dashboard/pull/5678

For the service count, depends on https://github.com/epinio/epinio/issues/1471. It will be commented for now.

### CODE REVIEW

**Comment by scures** [comment]
## Old
image

## New
image

File: pkg/epinio/detail/applications.vue [hunk 1]
```diff
@@ -125,73 +194,51 @@ export default Vue.extend<Data, any, any, any>({
               </thead>
               <tr>
                 <td>{{ t('tableHeaders.memory') }}</td>
-                <td>{{ value.instanceMemory.min }}</td>
-                <td>{{ value.instanceMemory.max }}</td>
-                <td>{{ value.instanceMemory.avg }}</td>
+                <td>{{ resourcesUssage.memoryBytes.toFixed(2) }} MiB</td>
```
Comment (richard-cox) [inline pkg/epinio/detail/applications.vue:197]: The memory and cpu stats were already calculated in the application model. Can't remember if that was an older change or not, but we should probably use that version

File: pkg/epinio/detail/applications.vue [hunk 2]
```diff
@@ -51,70 +49,141 @@ export default Vue.extend<Data, any, any, any>({
...
-          <h1>{{ value.configCount }}</h1>
-          <h3>
-            {{ t('epinio.applications.detail.counts.config') }}
-          </h3>
+      </template>
+
+      <template v-slot:top-right>
+        <!-- // TODO: Depends on https://github.com/epinio/epinio/issues/1345 -->
+        <!-- <span>Uptime 1d 23min</span>
+        Age: 2d ago -->
```
Comment (richard-cox) [inline pkg/epinio/detail/applications.vue:115]: The creation time has now been added(in meta.createdAt). Unfortunately I don't think we're getting uptime anytime soon

File: pkg/epinio/detail/applications.vue [hunk 3]
```diff
@@ -51,70 +49,141 @@ export default Vue.extend<Data, any, any, any>({
...
+      </template>
+
+      <!-- Resources count slot -->
+      <template v-slot:resourcesCount>
+        <!-- // TODO: Depends on https://github.com/epinio/epinio/issues/1471 -->
+        <!-- <div>
+          {{ value.configCount }} {{ t('epinio.applications.detail.counts.config') }}
+        </div> -->
+        <div>
+          {{ value.configCount }} {{ t('epinio.applications.detail.counts.services') }}
```
Comment (richard-cox) [inline pkg/epinio/detail/applications.vue:125]: There's `Configurations` and there's also `Services` (which are instances of `Catalog Services`). The app can be bound to both, however the `configurations` property currently contains resources that apply to both types (as part of binding a service to an application a configuration is counted). Until https://github.com/epinio/epinio/issues/1471 is created we should keep this labelled as `Configurations`

File: pkg/epinio/detail/applications.vue [hunk 4]
```diff
@@ -51,70 +49,141 @@ export default Vue.extend<Data, any, any, any>({
...
 
-  computed: {}
-
+        return acc;
+      }, {
+        millicpus:    0,
+        memoryBytes: 0,
+      });
+    },
+    choseIcon() {
```
Comment (richard-cox) [inline pkg/epinio/detail/applications.vue:74]: This isn't used?

File: pkg/epinio/detail/applications.vue [hunk 5]
```diff
@@ -51,70 +49,141 @@ export default Vue.extend<Data, any, any, any>({
...
+      return this.value.metadata.icon || 'fa-cube';
+    },
+    setIconType() {
+      switch (this.value.sourceInfo?.label) {
```
Comment (richard-cox) [inline pkg/epinio/detail/applications.vue:78]: Switching on human text generally isn't advisable. The sourceInfo under the hood uses ` switch (this.origin.Kind) { 
    case APPLICATION_MANIFEST_SOURCE_TYPE.PATH:`
...

However, would advise adding an `icon` property to that `sourceInfo` object. To catch the blank case you can just use something like `value.sourceInfo.icon` || `icon-epinio`

File: pkg/epinio/detail/applications.vue [hunk 6]
```diff
@@ -51,70 +49,141 @@ export default Vue.extend<Data, any, any, any>({
...
+          >
+          </ConsumptionGauge>
+          <div class="scale-instances">
+            <PlusMinus class="mt-15 mb-10" :value="value.desiredInstances" :disabled="saving" @minus="updateInstances(value.desiredInstances - 1)" @plus="updateInstances(value.desiredInstances + 1)" />
+          </div>
+        </SimpleBox>
+        <!-- Source information -->
+        <SimpleBox v-if="value.sourceInfo">
+          <div class="deployment__origin__row">
+            <h4>Deployment Details</h4>
```
Comment (richard-cox) [inline pkg/epinio/detail/applications.vue:155]: The switch to ul/li leaves the spacing unable to consume the correct amounts

image

File: pkg/epinio/detail/applications.vue [hunk 7]
```diff
@@ -5,20 +5,20 @@ import SimpleBox from '@shell/components/SimpleBox.vue';
 import ConsumptionGauge from '@shell/components/ConsumptionGauge.vue';
 import { EPINIO_PRODUCT_NAME, EPINIO_TYPES } from '../types';
 import ResourceTable from '@shell/components/ResourceTable.vue';
-import Tabbed from '@shell/components/Tabbed/index.vue';
-import Tab from '@shell/components/Tabbed/Tab.vue';
 import PlusMinus from '@shell/components/form/PlusMinus.vue';
 import { epinioExceptionToErrorsArray } from '../utils/errors';
-
+import ApplicationCard from '~/shell/components/cards/ApplicationCard.vue';
```
Comment (richard-cox) [inline pkg/epinio/detail/applications.vue:10]: There's a frustrating issue that VSCode will default to imports with `~` when we don't want them to. For instance this should be `@shell`

**Review by richard-cox — CHANGES_REQUESTED** [review]
After rebasing you'll need to use the latest epinio, i need to update the public instance with this and will let you know when it's done.

**Pull request merged**